tests/acceptance/standard/test_hooks_processor.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import pytest
+import textwrap
+import time
+from typing import TYPE_CHECKING
+
+import gitlab
+from gitlab.v4.objects import ProjectHook, Project, Group
+
+from gitlabform import GitLabForm
+from gitlabform.gitlab import GitlabWrapper
+from gitlabform.gitlab.core import NotFoundException
+
+from tests.acceptance import allowed_codes, run_gitlabform, create_group, create_project
+
+logger = logging.getLogger(__name__)
+
+
+def create_test_hook(project: Project):
+    hook = project.hooks.create(
+        {
+            "url": "http://birdman.chirp/update",
+            "push_events": False,
+            "merge_requests_events": False,
+        }
+    )
+    if TYPE_CHECKING:
+        assert isinstance(hook, ProjectHook)
+    return hook
+
+
+def prepare_project(gl, group: str, project: str):
+    with allowed_codes(404):
+        gl.groups.delete(group)
+        # wait for delete to finish
+        time.sleep(6)
+    created_group: Group = create_group(group)
+    created_project: Project = create_project(created_group, project)
+    return
+
+
+def reset_gitlab(gl):
+    with allowed_codes(404):
+        gl.groups.delete("test_group")
+        time.sleep(5)
+    return
+
```

Comment:
Shouldn't need these methods (i.e. `prepare_project`, `reset_gitlab`). There are some common "pytest fixtures" already in `tests/acceptance/conftest.py` file. You simply add those fixtures in your test method as parameter. Depending on the lifecycle of the fixtures, they will get cleaned up automatically.

Check out the other test files at `tests/acceptance/standard/` directory.

Reply:
yep, understood